python/tchannel/tornado/peer.py

```diff
@@ -525,160 +524,124 @@ def send(self, arg1, arg2, arg3,
         for k, v in self.headers.iteritems():
             headers.setdefault(k, v)
 
-        peer = self.peer
-        connection = yield peer.connect()
-
-        request = Request(
-            service=self.service,
-            argstreams=[InMemStream(endpoint), arg2, arg3],
-            id=connection.next_message_id(),
-            headers=headers,
-            endpoint=endpoint,
-            ttl=ttl,
-            tracing=Trace(
-                name=endpoint,
-                trace_id=trace_id,
-                parent_span_id=parent_span_id,
-                endpoint=Endpoint(self.peer.host,
-                                  self.peer.port,
-                                  self.service),
-                traceflags=traceflag,
-            )
-        )
-
         # only retry on non-stream request
-        if request.is_streaming_request or self._hostport:
+        if Request.is_streaming_request(arg2, arg3):
```

Comment:
This is ignoring the cached `_is_streaming_request` now? That's an expensive function call to make, this is probably a performance regression.